alicloud/data_source_alicloud_images.go

```diff
@@ -273,6 +277,11 @@ func imagesDescriptionAttributes(d *schema.ResourceData, images []ecs.ImageType,
 	if err := d.Set("images", s); err != nil {
 		return err
 	}
+
+	// create a json file in current directory and write data source to it.
+	if output, ok := d.GetOk("output_file"); ok && output != nil {
```

Comment:
Does this work as shown? `output_file` is `schema.TypeString`, so the `output` of d.GetOk` will be `""` . I would imagine this would result in a panic at runtime trying to compare a `string` to `nil`. 

I think instead you want to check for `""` instead of `nil`, does that sound right?

Reply:
@catsby Thanks for your review and suggestion. 

Actually, there is no error when I run the data source, but above code is really has a problem. I have improved it and please review again, thanks a lot.

Regards,
Guimin He